doc/stdenv/passthru.chapter.md

```diff
@@ -94,7 +94,7 @@ stdenv.mkDerivation (finalAttrs: {
   installCheckPhase = ''
     runHook preInstallCheck
     echo checking if 'git --version' mentions ${finalAttrs.version}
-    $out/bin/git --version | grep ${finalAttrs.version}
+    env --ignore-environment $out/bin/git --version | grep --silent "${finalAttrs.version}"
```

Comment:
Is this change solving for a specific failure mode in the example, or is it just demonstrating one way to guard against state inside the build environment in order to make a general point about the fact that tests run during the build may be sensitive to temporary state that won't exist after the build completes?

If it's the former, I think it might be better to just lay the cards down here and show the problem we'd have without this workaround (because it will provide a better illustration of the underlying risk of these tests depending on the state inside the sandbox) and also show the fix?

If this just demonstrates how to guard against state without an acute need, I think it's still worth explaining why a "clean" environment is important (and perhaps noting that this only guards against environment variables which are only one kind of state present in  the sandbox).

Reply:
This is specifically to avoid variables like `PYTHONPATH`, to guard against build artifacts one could `pushd "$(mkdtemp -d)"`